modules/nlu/src/backend/training-pipeline.ts

```diff
@@ -530,51 +524,32 @@ export const Trainer: Trainer = async (input: TrainInput, tools: Tools): Promise
     normalizedProgress = scaledProgress
     debouncedProgress(input.botId, 'Training', { ...input.trainingSession, progress: normalizedProgress })
   }
-  try {
-    let output = await PreprocessInput(input, tools)
-    output = await TfidfTokens(output)
-    output = ClusterTokens(output, tools)
-    output = await ExtractEntities(output, tools)
-    output = await AppendNoneIntent(output, tools)
-    const exact_match_index = BuildExactMatchIndex(output)
-    reportProgress()
-    const [oos_model, ctx_model, intent_model_by_ctx, slots_model] = await Promise.all([
-      TrainOutOfScope(output, tools, reportProgress),
-      TrainContextClassifier(output, tools, reportProgress),
-      TrainIntentClassifier(output, tools, reportProgress),
-      TrainSlotTagger(output, tools, reportProgress)
-    ])
-
-    const artefacts: TrainArtefacts = {
-      list_entities: output.list_entities,
-      oos_model,
-      tfidf: output.tfIdf,
-      ctx_model,
-      intent_model_by_ctx,
-      slots_model,
-      vocabVectors: output.vocabVectors,
-      exact_match_index
-      // kmeans: {} add this when mlKmeans supports loading from serialized data,
-    }
-
-    _.merge(model, { success: true, data: { artefacts, output } })
-  } catch (err) {
-    if (err instanceof TrainingCanceledError) {
-      debugTraining.forBot(input.botId, 'Training aborted')
-    } else {
-      // TODO use bp.logger once this is moved in Engine2
-      console.log('Could not finish training NLU model', err)
-    }
-    model.success = false
-  } finally {
-    model.finishedAt = new Date()
-    return model as Model
-  }
-}
 
-class TrainingCanceledError extends Error {
-  constructor() {
-    super('Training cancelled')
-    this.name = 'CancelError'
+  let stepOutput = await PreprocessInput(input, tools)
```

Comment:
personnal pref, but I would call this variable only `step` not `stepOutput`.

Reply:
Changed, not coherent with predict-pipeline but pref too